src/core/connection.c

```diff
@@ -2025,6 +2029,11 @@ QuicConnHandshakeConfigure(
             Connection->OrigCID = NULL;
         }
 
+        if (Connection->Datagram.Enabled) {
+            LocalTP.Flags |= QUIC_TP_FLAG_MAX_DATAGRAM_FRAME_SIZE;
+            LocalTP.MaxDatagramFrameSize = QUIC_DEFAULT_MAX_DATAGRAM_LENGTH;
+        }

```

Comment:
This could be moved outside the `if(Server)... else ...` blocks to reduce duplication.

Reply:
We could, but I prefer to keep this design, as it's used for all our TP config.